From eed0dc1cc32c2d251208e0283ec9afa63528e600 Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Wed, 17 Feb 2021 19:55:35 +0200
Subject: [PATCH] [SW-20154] change keys location offset in efuse

Key hashes are located at different efuse offsets
in Gaudi2/Greco. Hence make it platform specific

Change-Id: Ifce023ab758caa04651815f37d5d1366c3b53e19
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/common/efuse.h b/common/efuse.h
index 2915380..08c71ed 100644
--- a/common/efuse.h
+++ b/common/efuse.h
@@ -16,7 +16,6 @@
 
 #define EFUSE_READ_RANGE_ALL			0x3F80
 
-#define EFUSE_KEY_HASH_OFFSET			0x12
 #define EFUSE_SYSCFG_OFFSET			0x3A
 #define EFUSE_SECURITY_FW_VERSION_OFFSET	0x3B
 
diff --git a/gaudi2_arc_gnu/dev_efuse.h b/gaudi2_arc_gnu/dev_efuse.h
index 4ff9982..9f5728e 100644
--- a/gaudi2_arc_gnu/dev_efuse.h
+++ b/gaudi2_arc_gnu/dev_efuse.h
@@ -3,6 +3,7 @@
  * Copyright (C) 2021 HabanaLabs Ltd.
  */
 
+#define EFUSE_KEY_HASH_OFFSET				0x14
 #define EFUSE_SYSCFG_RECOVERY_CODE_LOCATION		13
 #define EFUSE_SYSCFG_RECOVERY_CODE_LOCATION_MASK	0x1
 #define EFUSE_SYSCFG_ENABLE_TPM_MASK			BIT(11)
diff --git a/goya2_arc/dev_efuse.h b/goya2_arc/dev_efuse.h
index bc1caf5..11a630a 100644
--- a/goya2_arc/dev_efuse.h
+++ b/goya2_arc/dev_efuse.h
@@ -3,6 +3,7 @@
  * Copyright (C) 2021 HabanaLabs Ltd.
  */
 
+#define EFUSE_KEY_HASH_OFFSET			0x12
 #define EFUSE_SYSCFG_ENABLE_4B_TERMINATION	BIT(25)
 #define EFUSE_SYSCFG_FW_VERSION_LOCATION	13
 #define EFUSE_SYSCFG_FW_VERSION_LOCATION_MASK	0x3
